src/Controller/ViewerController.php

```diff
@@ -125,6 +129,40 @@ public function editor(MediaInterface $media, Request $request, $edit = FALSE):
     return new Response((string) $this->renderer->renderRoot($render_array));
   }
 
+  /**
+   * Reads the 'destination' parameter from a request.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   Incoming request, which or may not have a 'destination' query parameter.
+   *
+   * @return \Drupal\Core\Url|null
+   *   The 'destination' as a url, or NULL if no destination was provided, or it
+   *   was empty.
+   *
+   * @see \Drupal\Core\Form\ConfirmFormHelper::buildCancelLink()
+   * @see \Drupal\Core\EventSubscriber\RedirectResponseSubscriber::getDestinationAsAbsoluteUrl()
+   */
```

Comment:
It would have been more straightforward to simply read the destination parameter with javascript. However, I have not found this done in Drupal core.\
Reading the parameter in php makes sure that we apply the same processing as in other places.

This said, I did look for other places that parse the destination parameter, and it does not look very standardized.